src/wrapper.rs

```diff
@@ -582,6 +594,55 @@ impl DxcValidator {
     }
 }
 
+pub struct Reflection {
+    inner: ComPtr<ID3D12ShaderReflection>,
+}
+impl Reflection {
+    fn new(inner: ComPtr<ID3D12ShaderReflection>) -> Self {
+        Self { inner }
+    }
+
+    pub fn inner_ptr(&self) -> *const ID3D12ShaderReflection {
+        self.inner.as_ptr()
+    }
```

Comment:
Fine by me to expose `ID3D12ShaderReflection` for now instead of writing a bunch of wrappers, but perhaps `&ID3D12ShaderReflection` or `NonNull<ID3D12ShaderReflection>` is a bit more ergonomic to use?

Alternatively wrap just the functions we need such as `get_thread_group_size`, its API is terribly non-Rusty anyway :)

Reply:
Good point. I opted for the `thread_group_size` getter.